servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -413,6 +423,19 @@ private void removeBlockNoLock(long userId, long blockId) throws IOException {
     mMetaManager.removeBlockMeta(blockMeta);
   }
 
+  private void updateLockedBlocks() {
+    mLockedBlocks.clear();
+    mLockedBlocks.addAll(mLockManager.getLockedBlocks());
+  }
+
+  // always update the metadata view before doing freeSpacewithNewView
+  private BlockMetadataManagerView getUpdatedView() throws IOException {
+    // TODO: update the view object instead of creating new one every time
+    updateLockedBlocks();
+    mMetadataView = new BlockMetadataManagerView(mMetaManager, mPinnedInodes, mLockedBlocks);
```

Comment:
The thing is we don't know whether we will make the view long-living: if the view is only short living, then we don't even need mMetadataView; if it is long living, others might access the view without doing the update.